templates/typescript_gapic/src/$version/$service_client.ts.njk

```diff
@@ -466,23 +478,35 @@ export class {{ service.name }}Client {
     }
     {{ util.buildHeaderRequestParam(method) }}
     this.{{ id.get("initialize") }}();
+    {%- if method.options and method.options.deprecated %}
+    gax.warn('DEP1001','{{method.name}} is deprecated and may be removed in a future version.', 'DeprecationWarning');
```

Comment:
Same here, maybe `DEP${service.name}-${method.name}` or something?

Reply:
BTW, what's the DEP 1001 and above DEP 1000? Maybe I miss something here.

Reply:
I just made up the ID so there was a separate ID for a deprecated service and a deprecated method. But to Alex's point, they would not be unique for different services or methods, so using the service/method name in the ID itself makes more sense.